src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -260,7 +259,40 @@ synchronized int insertRow(String tableName, List<Object> row) throws SQLExcepti
       // We always expect one row whether we query for the latest row or we query for a row by the rowID.
       throw new IllegalStateException("Expected one row, found: '" + recordList + "'");
     }
-    Record record = recordList.get(0);
+    return readFields(clz, recordList.get(0), tableName);
+  }
+
+  @Override
+  public synchronized <T> @org.checkerframework.checker.nullness.qual.Nullable List<T> readLatestGroup(Class<T> clz)
```

Comment:
Should we call it `action` group ?

Reply:
This API seems generic and not specific to actions. we could call it `readLatestEntrySet` ?

Reply:
Renamed it to `readAllForMaxTimeStamp`